src/modules/statsd.c

```diff
@@ -285,6 +298,55 @@ statsd_handle_payload(noit_check_t **checks, int nchecks,
   }
 }
 
+static inline void
+statsd_handle_one_entry(addr_t *addr, socklen_t addrlen,
+                        noit_check_t *parent, char *payload,
+                        noit_module_t *self) {
+  char ip[INET6_ADDRSTRLEN];
+  noit_check_t *checks[MAX_CHECKS];
+  int nchecks = 0;
+  switch(addr->in.sin_family) {
+    case AF_INET:
+      addrlen = sizeof(struct sockaddr_in);
+      inet_ntop(AF_INET, &((struct sockaddr_in *)addr)->sin_addr, ip, addrlen);
+      break;
+    case AF_INET6:
+      addrlen = sizeof(struct sockaddr_in6);
+      inet_ntop(AF_INET6, &((struct sockaddr_in6 *)addr)->sin6_addr, ip, addrlen);
+      break;
+    default:
+      ip[0] = '\0';
+  }
+  if(*ip) {
+    nchecks = noit_poller_lookup_by_ip_module(ip, self->hdr.name, checks, MAX_CHECKS-1);
+  }
+  nchecks += noit_poller_lookup_by_ip_module("0.0.0.0", self->hdr.name,
+    checks+nchecks, MAX_CHECKS-1-nchecks);
+  if(parent) {
+    checks[nchecks++] = parent;
+  }
+  if(nchecks) {
+    statsd_handle_payload(checks, nchecks, payload, strlen(payload));
```

Comment:
This calls into `update_check` which does seem to be thread safe.  It seems individual threads would update the same data concurrently

Reply:
You say it does semn to be thread safe. I question that assertion. I believe that is incorrect.